### ISSUE TITLE
compile with MSVC 

### ISSUE THREAD

**Issue by gadget114514** [yasl-lang/yasl#101]
Just information

```
#define TOKEN_MATCHES(parser, ...)  (CHOOSE(__VA_ARGS__, T4, T3, T2, T1, T0)(parser, __VA_ARGS__))
```
macro in parser.h, MSVC cannot treat the macro correctly.
I tried to make work around by writing following code, it does look well on IDE but MSVC compiler cannot compile it.
So in case of porting to MSVC, some trick will be required.

```

#define VA_NARGS(...) VA_NARGS_IMPL_((__VA_ARGS__, 5, 4, 3, 2, 1))
#define VA_NARGS_IMPL_(tuple) VA_NARGS_IMPL tuple
#define VA_NARGS_IMPL(_1, _2, _3, _4, _5, N, ...) N
#define FULLY_EXPANDED(count, tuple) T##count tuple
#define SEMI_EXPANDED(count, ...) FULLY_EXPANDED(count, (parser, __VA_ARGS__))
#define EXPAND_THESE(...) SEMI_EXPANDED(VA_NARGS(__VA_ARGS__), __VA_ARGS__)
#define T1(p, a) (curtok(p) == a)
#define T2(p, a, b) T1(p, a) || T1(p, b)
#define T3(p, a, b, c) T2(p, a, b) || T1(p, c)
#define T4(p, a, b, c, d) T3(p, a, b, c) || T1(p, d)
#define TOKEN_MATCHES(px, ...) \
  (SEMI_EXPANDED(VA_NARGS(__VA_ARGS__), __VA_ARGS__))
 
```

**Comment by gadget114514** [comment]
i used MSVC 2017 community version.

**Comment by gadget114514** [comment]
there is workaround.

**Comment by CoffeeTableEspresso** [comment]
merged this into v0.3.5, thanks for the help!
